sdk/core/System.ClientModel/README.md

```diff
@@ -23,71 +22,290 @@ dotnet add package System.ClientModel
+
+```C# Snippet:ClientResultExceptionReadme
+try
+{
+    IPAddress ipAddress = IPAddress.Parse("2001:4898:80e8:b::189");
+    ClientResult<IPAddressCountryPair> result = await client.GetCountryCodeAsync(ipAddress);
+}
+// Handle exception with status code 404
+catch (ClientResultException e) when (e.Status == 404)
+{
+    // Handle not found error
+    Console.Error.WriteLine($"Error: Response failed with status code: '{e.Status}'");
+}
+```
+
+### Configuring service clients
+
+Service clients provide a constructor that takes a service endpoint and a credential used to authenticate with the service.  They also provide a constructor overload that takes an endpoint, a credential, and an instance of `ClientPipelineOptions`.
+Passing `ClientPipelineOptions` when a client is created will configure the pipeline that the client uses to send and receive HTTP requests and responses.  Client pipeline options can be used to override default values such as the network timeout used to send or retry a request.
+
+```C# Snippet:ClientModelConfigurationReadme
+MapsClientOptions options = new()
+{
+    NetworkTimeout = TimeSpan.FromSeconds(120),
+};
+
+string? key = Environment.GetEnvironmentVariable("MAPS_API_KEY");
+ApiKeyCredential credential = new(key!);
+MapsClient client = new(new Uri("https://atlas.microsoft.com"), credential, options);
 ```
 
-### Read and write persistable models
+For more information on client configuration, see [Client configuration samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/Configuration.md).
+
+### Customizing HTTP requests
+
+Service clients expose low-level [protocol methods](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/ServiceMethods.md#protocol-methods) that allow callers to customize HTTP requests by passing an optional `RequestOptions` parameter.  `RequestOptions` can be used to modify various aspects of the request sent by the service method, such as adding a request header, or adding a policy to the client pipeline that can modify the request directly before sending it to the service. `RequestOptions` also allows an client user to pass a `CancellationToken` to the method.
 
-As a library author you can implement `IPersistableModel<T>` or `IJsonModel<T>` which will give library users the ability to read and write your models.
+```C# Snippet:RequestOptionsReadme
+// Create RequestOptions instance
+RequestOptions options = new();
 
-Example writing an instance of a model.
+// Set CancellationToken
+options.CancellationToken = cancellationToken;
 
-```C# Snippet:Readme_Write_Simple
-InputModel model = new InputModel();
-BinaryData data = ModelReaderWriter.Write(model);
+// Add a header to the request
+options.AddHeader("CustomHeader", "CustomHeaderValue");
+
+// Call protocol method to pass RequestOptions
+ClientResult output = await client.GetCountryCodeAsync(ipAddress.ToString(), options);
```

Comment:
I think it would be good if this section showed a method that takes BodyContent and how to serialize a model to BodyContent.

Reply:
Cool - I have that in the next section, but I can merge the two.

Reply:
Added, and will keep the second section to expand on the different ways to create input content.